src/libraries/System.Private.CoreLib/src/System/Threading/LowLevelLock.cs

```diff
@@ -150,7 +154,7 @@ private void WaitAndAcquire()
             VerifyIsNotLocked();
 
             // Spin a bit to see if the lock becomes available, before forcing the thread into a wait state
-            if (_spinWaiter.SpinWaitForCondition(_spinWaitTryAcquireCallback, SpinCount, SpinSleep0Threshold))
+            if (_spinWaiter.SpinWaitForCondition(s_spinWaitTryAcquireCallback, this, SpinCount, SpinSleep0Threshold))
```

Comment:
Cannot this be simple code like below where the compiler generates the caching?

```c#
if (_spinWaiter.SpinWaitForCondition(SpinWaitTryAcquireCallback, this, SpinCount, SpinSleep0Threshold))
```

Reply:
If we want it to be lazily created, yes. Given the code was proactively allocating a delegate upon construction, I kept the eagerness. If Kount says lazy is better, I can change it.

Reply:
IIRC the eagerness was to avoid allocation on this path, such that it couldn't fail here